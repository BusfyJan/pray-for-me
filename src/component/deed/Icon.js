import React, { Component } from "react";
import MassIcon from "material-ui-icons/Home";
import PrayerIcon from "material-ui-icons/ChatBubbleOutline";
import { green, blue } from "material-ui/colors";
import styled from "styled-components";

class Icon extends Component {
    render() {
        let Icon = null;
        let color = null;

        if (this.props.type === "prayer") {
            Icon = PrayerIcon;
            color = green[500];
        }

        if (this.props.type === "mass") {
            Icon = MassIcon;
            color = blue[500];
        }

        if (Icon === null) {
            throw new Error("Icon type " + this.props.type + " not supported!");
        }

        Icon = styled(Icon)`
            color: ${color};
        `;

        return <Icon />;
    }
}

export default Icon;
