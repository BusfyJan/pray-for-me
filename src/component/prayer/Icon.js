import React, { Component } from "react";
import HealthIcon from "material-ui-icons/Accessibility";
import MoneyIcon from "material-ui-icons/AttachMoney";
import RelationshipsIcon from "material-ui-icons/Favorite";
import TimeIcon from "material-ui-icons/Watch";
import GuidanceIcon from "material-ui-icons/Gesture";
import JoyIcon from "material-ui-icons/InsertEmoticon";
import PeaceIcon from "material-ui-icons/FilterVintage";
import MotivationIcon from "material-ui-icons/Whatshot";
import SchoolAndJobIcon from "material-ui-icons/School";
import {
    green,
    blue,
    red,
    yellow,
    orange,
    deepPurple,
    pink
} from "material-ui/colors";
import styled from "styled-components";

class Icon extends Component {
    render() {
        let Icon = null;
        let color = null;

        if (this.props.type === "guidance") {
            Icon = GuidanceIcon;
            color = blue[500];
        }

        if (this.props.type === "health") {
            Icon = HealthIcon;
            color = pink[500];
        }

        if (this.props.type === "joy") {
            Icon = JoyIcon;
            color = yellow[500];
        }

        if (this.props.type === "money") {
            Icon = MoneyIcon;
            color = green[500];
        }

        if (this.props.type === "peace") {
            Icon = PeaceIcon;
            color = deepPurple[500];
        }

        if (this.props.type === "relationships") {
            Icon = RelationshipsIcon;
            color = red[500];
        }

        if (this.props.type === "time") {
            Icon = TimeIcon;
            color = orange[500];
        }

        if (this.props.type === "schoolAndJob") {
            Icon = SchoolAndJobIcon;
            color = blue[500];
        }

        if (this.props.type === "motivation") {
            Icon = MotivationIcon;
            color = deepPurple[500];
        }

        if (Icon === null) {
            throw new Error("Icon type " + this.props.type + " not supported!");
        }

        Icon = styled(Icon)`
            color: ${color};
        `;

        return <Icon />;
    }
}

export default Icon;
