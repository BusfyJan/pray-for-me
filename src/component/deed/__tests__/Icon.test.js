import React from 'react'
import Icon from '../Icon'
import renderer from 'react-test-renderer'

test('Icon render', () => {
  const component = renderer.create(<Icon type="mass" />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
