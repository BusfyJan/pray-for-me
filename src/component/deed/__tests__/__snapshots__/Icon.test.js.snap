// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render 1`] = `
<svg
  aria-hidden="true"
  className="MuiSvgIcon-root-1 sc-bdVaJa iehYsS"
  color={undefined}
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
  />
</svg>
`;
